src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -123532,6 +123532,56 @@ static void M1(C c, string? s)
                     Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(18, 15));
         }
 
+        [Fact, WorkItem(42960, "https://github.com/dotnet/roslyn/issues/42960")]
+        public void EqualsMethod_IncompleteBaseClause()
+        {
+            var source = @"
+class C :
+{
+    void M(C? other)
+    {
+        if (Equals(other)) { other.ToString(); }
```

Comment:
>Equals(other) [](start = 12, length = 13)

The relationship between the test scenario and the implementation is not obvious. The implementation checks if we have a receiver, what is the relationship with the base type and the lack of receiver?

Reply:
I find it surprising as well. It seems like the erroneous base clause causes us to bind the call differently, with a null ReceiverOpt.

I did find that the crash was not produced when calling `this.Equals(other)`. So maybe that would be illustrative to add to the test along with some comment. Or maybe we should go looking in the binder and decide whether the behavior that's happening now is a bug in binding.

Reply:
> I find it surprising as well. It seems like the erroneous base clause causes us to bind the call differently, with a null ReceiverOpt.

It is quite possible that binder simply creates an erroneous call node in this case, which doesn't really represent anything that we can learn from (perhaps even the method symbol is an ErrorMethodSymbol). Is this the case? I think IOperation factory detects nodes like that and doesn't even surface them as calls. Consider if we should use the same kind of detection in the nullable walker and bail even earlier than the null check that you added.

---
In reply to: [401748792](https://github.com/dotnet/roslyn/pull/42968#discussion_r401748792) [](ancestors = 401748792)